### PR TITLE
DEV-1110: Simplify roomConfig check

### PR DESCRIPTION
**Pull request by kimberleejohnson**
@lizadaily called out how we could simplify the code that checks for a roomConfig, so this PR implements that! 

Tested with a cloud-beta recording and everything still works!

### CODE REVIEW

**Review by ablwr — APPROVED** [review]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1:

File: custom/shared/contexts/CallProvider.js [hunk 1]
```diff
@@ -52,9 +52,10 @@ export const CallProvider = ({
...
     const updateRoomConfigState = async () => {
       const roomConfig = await daily.room();
-      if (!('config' in roomConfig)) return;
+      const config = roomConfig?.config;
+      if (!config) return;
 
-      if (roomConfig?.config?.exp) {
+      if (config.exp) {
         setRoomExp(
           roomConfig?.config?.exp * 1000 || Date.now() + 1 * 60 * 1000
```
Comment (lizadaily) [inline custom/shared/contexts/CallProvider.js:60]: Could also reuse the `config` var here vs retrieving it again I think, since we know it's available!

Comment (kimberleejohnson) [reply custom/shared/contexts/CallProvider.js]: @lizadaily , great catch, thank you so much! I just updated `setRoomExp` to use the variable. Let me know if that looks better to you!

Comment (lizadaily) [reply custom/shared/contexts/CallProvider.js]: Nice! It does look better this way imo, but also I suspect limiting use of optional chaining will usually result in quicker execution. Did a quick benchmark test here: https://www.measurethat.net/Benchmarks/Show/16207/0/optional-chaining-vs-reuse
(Maybe for the purpose of an example it isn’t really a big impact, but seems like we may as well avoid  unnecessary overheard where possible)

**Review by lizadaily — APPROVED** [review]
🎉

**Pull request merged**